### PR TITLE
MRG, MAINT: Improve orientation tests for LCMV and DICS

### PR DESCRIPTION
**Pull request by larsoner**
1. Improve the LCMV orientation test to test only the orientations of the subset of vertices that were correctly localized.
2. Add DICS equivalents of the LCMV tests.

@britta-wstnr I think this might help with #9215

### CODE REVIEW

**Comment by larsoner** [comment]
@britta-wstnr can you look and we can merge? I think this is a useful testing enhancement

File: mne/beamformer/tests/test_dics.py [hunk 1]
```diff
@@ -738,15 +740,61 @@ def test_localization_bias_free(bias_params_free, reg, pick_ori, weight_norm,
...
+    loc = np.abs(apply_dics(evoked, filters).data)
+    ori = filters['max_power_ori'][0]
+    assert ori.shape == (246, 3)
+    loc = np.abs(loc)
+    # Compute the percentage of sources for which there is no loc bias:
+    max_idx = np.argmax(loc, axis=0)
+    mask = want == max_idx  # ones that localized properly
+    perc = mask.mean() * 100
+    assert lower <= perc <= upper
+    # Compute the dot products of our forward normals and
```
Comment (britta-wstnr) [inline mne/beamformer/tests/test_dics.py:783]: `and ....?` cliffhanger!

**Review by britta-wstnr — COMMENTED** [review]
I am still puzzled that agreement is so low. Both `fwd` models are surface only, right?

**Comment by larsoner** [comment]
@britta-wstnr indeed it's disappointing that it's so low, not sure why. But this at least makes the test clearer/cleaner so I think it's a step in the right direction. Okay to merge once CIs come back happy (and I update the comment)?

**Comment by larsoner** [comment]
Okay with you @britta-wstnr ?

**Comment by britta-wstnr** [comment]
Yes, I agree it makes things easier to read.
I've been trying to think why we get such bad results, but I did not come up with anything useful ..... So fine with me to merge here, although it would be great to figure out eventually why this is so low!

**Pull request merged**